app/controllers/atmosphere/api/v1/appliance_types_controller.rb

```diff
@@ -93,6 +78,14 @@ def endpoint_payload
 
         private
 
+        def check_for_conflict!(appl)
+          vm = appl.virtual_machines.first
+          if vm && vm.state.saving?
+            raise Atmosphere::Conflict,
+                  'It is not allowed to save application twice'
```

Comment:
This is original message, I did not change here anything. If you believe this message should be improved please do it in another PR.